src/srldrvce/srldrvce.h

```diff
@@ -77,10 +110,14 @@ typedef struct {
     uint8_t tx_addr;
     srl_device_type_t type;
     srl_device_subtype_t subtype;
+    uint16_t idvendor;
+    uint16_t idproduct;
+    uint16_t bcddevice;
+    uint8_t baudtype;
     ring_buf_ctrl_t rx_buf;
     ring_buf_ctrl_t tx_buf;
     srl_error_t err;
-    uint8_t reserved[16];
```

Comment:
I'm a little worried about this change - does this break out backwards compatibility when user allocated structures might be broken? @commandblockguy 

The issue is someone using a program compiled for an old version of the lib, and using a newer lib.

Reply:
Could new fields be added at the end instead, and the reserved array shortened?

Reply:
This is why the structure should be big enough to support forwards compatibility and the addition of fields - also moving of fields breaks backwards compatibility for all programs too.

Reply:
@adriweb no, because the structure is user-allocated. so using a newer library would corrupt memory.

Reply:
Memory corruption could only happen if the total size of the struct changes, right?
In hindsight, yeah, it probably wasn't a good idea to make the internal structures visible - I mostly only did that to ease debugging, and didn't get around to removing them before actually releasing it.
Aside from the device field, I don't imagine that people have used the struct fields - it would probably be a good idea to deprecate them.